pombola/south_africa/views/person.py

```diff
@@ -243,6 +244,25 @@ def get_latest_meeting_urls(self, data):
     #  LDE: Arrived Late and Departed Early
     #  P:   Present
 
+    def get_position_for_year(self, year):
+        # Check if the Person held an active ministerial position the last day
+        # of the year. Return 'minister' if so, 'mp' if not.
+
+        year_end_date = datetime.date(year, 12, 31)
+        today = datetime.date.today()
+
+        active_minister = self.object.position_set.filter(
+                Q(title__slug__startswith='minister') |
+                Q(title__slug__startswith='deputy-minister')
+            ).currently_active(
+                today if year_end_date > today else year_end_date)
```

Comment:
Since this is similar to filtering logic used in your previous filter request, I think it might be good to factor these out into new methods on `pombola.core.models.PositionQuerySet`. e.g. you could have:

```python
    def at_end_of_year(self, year):
        ...
```

... and:

```python
    def title_slug_prefixes(self, title_slug_prefixes):
        qs = [Q(title_slug__startswith=prefix) for prefix in title_slug_prefixes]
        return self.filter(reduce(lambda acc, item: acc | item, qs)
```

(or something like that - I haven't tested that) and chain those methods. What do you think about that?